gradle-plugin/build.gradle

```diff
@@ -13,35 +13,39 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 plugins {
   id 'java-gradle-plugin'
   id 'groovy'
-  id 'maven'
+  id 'maven-publish'
 }
 
+apply from: "$rootDir/gradle/functional-test.gradle"
```

Comment:
Gradle file for test is a separate file. This follows the sample https://github.com/gradle/guides/blob/master/subprojects/testing-gradle-plugins/samples/code/url-verifier-plugin/build.gradle#L4